src/plugin/adapter-registry/c-adapter.ts

```diff
@@ -0,0 +1,33 @@
+/********************************************************************************
+ * Copyright (C) 2023 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import * as vscode from 'vscode';
+import * as manifest from '../manifest';
+import { VariableTracker } from './adapter-capabilities';
+import { outputChannelLogger } from '../logger';
+import { AdapterRegistry } from './adapter-registry';
+import { CTracker } from './c-tracker';
+
+export class CAdapter {
+    constructor(protected registry: AdapterRegistry) {
+    }
+
+    activate(context: vscode.ExtensionContext): void {
+        const debugTypes = vscode.workspace.getConfiguration(manifest.PACKAGE_NAME).get<string[]>(manifest.CONFIG_DEBUG_TYPES) || manifest.DEFAULT_DEBUG_TYPES;
```

Comment:
Would it be possible to add handling for changes in this configuration?

Reply:
Agreed, although the disposables chaining makes this slightly challenging :)

Reply:
Implemented as discussed